### DISCUSSION TITLE
Streaming content

### DISCUSSION THREAD

**Discussion by felipeAH** [jeremyevans/roda#268] [General]
Hi, very new to roda.

I'm trying to stream content with roda. 

    class SlowStreamer
      def each
        (1..10).each do |i|
          yield (i.to_s + "\n")
          sleep 1
        end
      end
    end

so i've tried to:

      ss = SlowStreamer.new
      response.finish_with_body ss

and get 

Roda::RodaError at /test
unsupported block result: [200, {"Content-Type"=>"text/html"}, #<Buttons::SlowStreamer:0x000055bd3a2206a8>]

what is the correct way to do it?

thank you very much.

Felipe.

**Comment by jeremyevans** [comment]
You probably want `request.halt response.finish_with_body(ss)`.

**Comment by felipeAH** [comment]
Worked fine. Thank you.

Jeremy Evans ***@***.***> writes:

> You probably want `request.halt response.finish_with_body(ss)`.


--
Felipe Alvarez Harnecker
 Tel: +56 9 9874 60 17
